systest/tests/equivocation_test.go

```diff
@@ -79,10 +82,31 @@ func TestEquivocation(t *testing.T) {
 			return true, nil
 		})
 	}
-	eg.Wait()
+	require.NoError(t, eg.Wait())
 	reference := results[cl.Client(0).Name]
 	for i := 1; i < cl.Total(); i++ {
 		assert.Equal(t, reference, results[cl.Client(i).Name],
-			"reference: %v, client: %v", cl.Client(0).Name, cl.Client(i).Name)
+			"reference: %v, client: %v", cl.Client(0).Name, cl.Client(i).Name,
+		)
 	}
+
+	// TODO(mafa): since the nodes running with shared keys are all perfectly synchronized, we can't
+	// detect equivocation directly. The two nodes sharing the same key will with high probability broadcast
+	// the same messages to other peers which won't trigger an equivocation
+	//
+	// For this test to work properly the nodes sharing keys need to be slightly out of sync from each other so
+	// they can broadcast different messages to the network.
```

Comment:
Perhaps the second nodes (the ones reusing keys) could join the network in a different epoch (so they create a different initial ATX) or use a different commitment ATX? In the former case they would be equivocated by "invalid previous ATX" and in the former - by "double ATX publish".

Reply:
I found a way to make the assertion work 🙂